src/packages/cli/src/cli.ts

```diff
@@ -38,109 +50,181 @@ const isDocker =
+    }
+  });
+} else if (argv.action === "stop") {
+  const instanceName = argv.name;
+
+  stopDetachedInstance(instanceName).then(instanceFound => {
+    if (instanceFound) {
+      console.log("Instance stopped");
+    } else {
+      console.error("Instance not found");
+    }
+  });
+} else if (argv.action === "start-detached") {
+  startDetachedInstance(process.argv, argv, version)
+    .then(instance => {
+      const highlightedName = porscheColor(instance.instanceName);
+      // output only the instance name to allow users to capture stdout and use to
+      // programmatically stop the instance
+      console.log(highlightedName);
     })
-    .on("SIGINT", () => {
-      // we must "close" the RL interface otherwise the process will think we
-      // are still listening
-      // https://nodejs.org/api/readline.html#readline_event_sigint
-      rl.close();
-      handleSignal("SIGINT");
+    .catch(err => {
+      // the child process would have output its error to stdout, so no need to
+      // output anything more
     });
-}
-
-process.on("SIGINT", handleSignal);
-process.on("SIGTERM", handleSignal);
-process.on("SIGHUP", handleSignal);
-
-async function startGanache(err: Error) {
-  if (err) {
-    console.log(err);
-    process.exitCode = 1;
-    return;
-  } else if (receivedShutdownSignal) {
-    closeHandler();
-    return;
-  }
-  started = true;
-
-  switch (flavor) {
-    case FilecoinFlavorName: {
-      await initializeFilecoin(
-        server.provider as FilecoinProvider,
-        cliSettings
+} else if (argv.action === "list") {
+  getDetachedInstances().then(instances => {
+    if (instances.length === 0) {
+      console.log(
+        `No detached instances found - try ${porscheColor(
+          "ganache --detach"
+        )} to start a detached instance`
       );
-      break;
-    }
-    case EthereumFlavorName:
-    default: {
-      initializeEthereum(server.provider as EthereumProvider, cliSettings);
-      break;
+    } else {
+      const now = Date.now();
+
+      const table = new Table({
+        head: [
+          chalk.bold("PID"),
+          chalk.bold("Name"),
+          chalk.bold("Flavor"),
+          chalk.bold("Version"),
+          chalk.bold("Host"),
+          chalk.bold("Port"),
+          chalk.bold("Uptime")
+        ],
+        style: {
+          head: ["white", "white", "white", "white", "white", "white", "white"]
+        }
+      });
+
+      instances.sort((a, b) => b.startTime - a.startTime);
+      for (let i = 0; i < instances.length; i++) {
+        const instance = instances[i];
+
+        const uptime = now - instance.startTime;
+        table.push([
+          instance.pid.toString(),
+          porscheColor(instance.instanceName),
+          instance.flavor,
+          instance.version,
+          instance.host,
+          instance.port.toString(),
+          formatUptime(uptime)
+        ]);
+      }
+
+      console.log(table.toString());
```

Comment:
Made an issue for the rendering issue here: https://github.com/trufflesuite/ganache/issues/3949